pkg/component/extensions/operatingsystemconfig/original/components/sshdensurer/templates/scripts/disable-sshd.tpl.sh

```diff
@@ -1,16 +1,12 @@
 #!/bin/bash -eu
 set -e
 
-# Disable sshd service if enabled
-if systemctl is-enabled --quiet sshd.service ; then
-    systemctl disable sshd.service
+# Disable and mask sshd service if not masked
+if [ "$(systemctl is-enabled sshd.service)" != "masked" ] ; then
+    systemctl disable sshd.service || true
+    systemctl mask --now sshd.service
 fi
 
-# Stop sshd service if active
-if systemctl is-active --quiet sshd.service ; then
-    systemctl stop sshd.service
-fi
-
-# Disabling the sshd service does not terminate already established connections
-# Kill all currently established ssh connections
-pkill -x sshd || true
+# Stopping the sshd service does not terminate already established connections
+# Kill all currently established orphaned ssh connections on the host
+pkill -P 1 -x sshd || true
```

Comment:
From my point of view the comment `Stopping the sshd service does not terminate...` should be reworked. In the script it is no longer obvious that sshd service is actually being stopped. You need implicit knowledge that the `--now` option of `systemctl mask` will stop a service. May that is also worth a comment.